src/pod.ts

```diff
@@ -65,6 +68,13 @@ export abstract class AbstractPod extends base.Resource {
     return [...this._hostAliases];
   }
 
+  public get podSelector(): PodSelector {
+    return {
+      labelSelector: Object.keys(this.podMetadata.toJson().labels).map(l => PodLabelQuery.is(l, this.podMetadata.getLabel(l)!)),
+      namespaceSelector: this.metadata.namespace ? [PodLabelQuery.is('kubernetes.io/metadata.name', this.metadata.namespace)] : undefined,
+    };
```

Comment:
Add unit test for this behavior (I was expecting to find `kubernetes.io/metadata.name` in one of the snapshots)

Reply:
Outdated, should be covered now.